src/repository/opamDownload.ml

```diff
@@ -287,7 +290,10 @@ module SWHID = struct
       None
 
   let get_dir hash =
-    let url = full_url "vault/directory" hash in
+    (* https://archive.softwareheritage.org/api/1/vault/flat/doc/ *)
+    let url = vault_url "flat" hash in
+    (* The POST is needed only for asking to cook the archive, it's a no-op on
+       status check *)
```

Comment:
I'm not sure what the purpose in the protocol of having `POST` and `GET` as separate verbs (I don't reading the docs see the purpose of the distiction), but could we not have `~post` being an argument for `get_dir` and then alter `get_url` below so that it passes `~post:true` the first time and `~post:false` for any retries (which would be following the protocol, even if it's not clear why the protocol says to do that)?

Reply:
From the protocol
> That endpoint [/vault/flat] enables to create a vault cooking task for a directory through a POST request or check the status of a previously created one through a GET request.

It is possible indeed to have a more robust mechanism:
1. first check with a GET request
  2. It is already cooked -> we get the json directly and go to 5.
  3. It is not cooked -> 404 error (exit 0 on curl and 8 on wget), and we need to retrieve the skipped body (curl -i, wget ?) to analyse the source (cooking not yet requested or object not found)
  4. Request cooking with a POST request
5. Analyse the returned json to retrieve status (and archive url)
6. iterate on 5 until it is cooked

Reply:
Is this what we're _meant_ to do, though? I don't quite understand why the protocol has bothered with the "dance" - if what we're presently doing of just always POSTing is OK, what's the purpose of having the behaviour of GET being different?

Reply:
This doesn't have to block merging, though, if we're sure that POSTing each time definitely isn't an issue.

Reply:
Update to remove all-PÖST requests